pkg/application/stateDiffUseCase.go

```diff
@@ -73,3 +83,52 @@ func (useCase *StateDiffUseCase) DetermineStateDiff(ctx context.Context, bluepri
 	// return it to signal that a repeated call to this function will not result in any progress.
 	return stateDiffError
 }
+
+func (useCase *StateDiffUseCase) collectEcosystemState(ctx context.Context, effectiveBlueprint domain.EffectiveBlueprint) (ecosystem.EcosystemState, error) {
+	logger := log.FromContext(ctx).WithName("StateDiffUseCase.collectEcosystemState")
+
+	//TODO: collect ecosystem state in parallel (like for ecosystem health) if we have time
+	//load current dogus and components
+	logger.Info("collect installed dogus")
+	installedDogus, doguErr := useCase.doguInstallationRepo.GetAll(ctx)
+	logger.Info("collect installed components")
+	installedComponents, componentErr := useCase.componentInstallationRepo.GetAll(ctx)
+	// load current config
+	logger.Info("collect needed global config")
+	globalConfig, globalConfigErr := useCase.globalConfigRepo.GetAllByKey(ctx, effectiveBlueprint.Config.Global.GetGlobalConfigKeys())
+	logger.Info("collect needed dogu config")
+	doguConfig, doguConfigErr := useCase.doguConfigRepo.GetAllByKey(ctx, effectiveBlueprint.Config.GetDoguConfigKeys())
+	logger.Info("collect needed sensitive dogu config")
+	sensitiveDoguConfig, sensitiveConfigErr := useCase.sensitiveDoguConfigRepo.GetAllByKey(ctx, effectiveBlueprint.Config.GetSensitiveDoguConfigKeys())
+
+	joinedError := errors.Join(doguErr, componentErr, globalConfigErr, doguConfigErr, sensitiveConfigErr)
+
+	var internalErrorType *domainservice.InternalError
```

Comment:
The `GetAllByKey`-methods can throw `NotFoundErrors` too. You have to handle this case.

Reply:
Ok i see that it is okay to ignore them. But should we check if the error is an error and no internal or notfound error even if the interface should only return both?

Reply:
To ignore NotFoundErrors is wanted behavior. There could be config keys in the blueprint which are not yet present in the ecosystem. These keys will throw `NotFoundErrors` but the repositories also give us the config entries for all other keys in combination with the joined errors. 

I added a comment to clarify this in the code.